### PR TITLE
update samuel/limit-uploaded-files to master and make changes to support Team Assignments

### PR DESCRIPTION
**Pull request by jansenk**
**TL;DR -** Pulled in master, and locked the option to True for team assignments.

**What changed?**

- Merged in master, updated tests that had diverged
- We decided, for now, that team assignments should always allow multiple files, so lock the option to True when the assignment is teams-enabled.
- When the upload fails because we are only allowing a single file, remove the file from the input, because otherwise we open ourselves up to allowing a "descriptionless" file

**_Since we have every missing master commit included in this merge comparison, definitely look just by commit. Look at my last couple of commits, thats where the real changes are._**

**Developer Checklist**
- [ ] Reviewed the [release process](./release_process.md)
- [ ] Translations up to date
- [ ] JS minified, SASS compiled
- [ ] Test suites passing on Jenkins
- [ ] Bumped version number in [setup.py](../setup.py) and [package.json](../package.json)

JIRA: [OSPR-3768](https://openedx.atlassian.net/browse/OSPR-3768)

FIY: @edx/masters-devs-gta

### CODE REVIEW

**Comment by jansenk** [comment]
Rather than this approach, I'm going to just merge this branch into edx:master
